opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -141,25 +141,34 @@
 class Resource:
     """A Resource is an immutable representation of the entity producing telemetry as Attributes."""
 
-    def __init__(self, attributes: Attributes):
+    def __init__(
+        self, attributes: Attributes, schema_url: typing.Optional[str] = ""
```

Comment:
Default values should be set to `None`. You can verify for None in the constructor and then set appropriately.